spec/engine/unit/segments/models_spec.rb

```diff
@@ -0,0 +1,173 @@
+  ['NOT_EQUAL', true, 'false', true],
+  ['NOT_EQUAL', false, 'true', true],
+  ['NOT_EQUAL', false, 'false', false],
+  ['NOT_EQUAL', true, 'true', false],
+  ['CONTAINS', 'bar', 'b', true],
+  ['CONTAINS', 'bar', 'bar', true],
+  ['CONTAINS', 'bar', 'baz', false],
+  ['NOT_CONTAINS', 'bar', 'b', false],
+  ['NOT_CONTAINS', 'bar', 'bar', false],
+  ['NOT_CONTAINS', 'bar', 'baz', true],
+  ['REGEX', 'foo', /[a-z]+/, true],
+  ['REGEX', 'FOO', /[a-z]+/, false],
+  ['REGEX', '1.2.3', /\d/, true],
+
+  ['EQUAL', '1.0.0', '1.0.0:semver', true],
+  ['EQUAL', '1.0.0', '1.0.1:semver', false],
+  ['NOT_EQUAL', '1.0.0', '1.0.0:semver', false],
+  ['NOT_EQUAL', '1.0.0', '1.0.1:semver', true],
+  ['GREATER_THAN', '1.0.1', '1.0.0:semver', true],
+  ['GREATER_THAN', '1.0.0', '1.0.0-beta:semver', true],
+  ['GREATER_THAN', '1.0.1', '1.2.0:semver', false],
+  ['GREATER_THAN', '1.0.1', '1.0.1:semver', false],
+  ['GREATER_THAN', '1.2.4', '1.2.3-pre.2+build.4:semver', true],
+  ['LESS_THAN', '1.0.0', '1.0.1:semver', true],
+  ['LESS_THAN', '1.0.0', '1.0.0:semver', false],
+  ['LESS_THAN', '1.0.1', '1.0.0:semver', false],
+  ['LESS_THAN', '1.0.0-rc.2', '1.0.0-rc.3:semver', true],
+  ['GREATER_THAN_INCLUSIVE', '1.0.1', '1.0.0:semver', true],
+  ['GREATER_THAN_INCLUSIVE', '1.0.1', '1.2.0:semver', false],
+  ['GREATER_THAN_INCLUSIVE', '1.0.1', '1.0.1:semver', true],
+  ['LESS_THAN_INCLUSIVE', '1.0.0', '1.0.1:semver', true],
+  ['LESS_THAN_INCLUSIVE', '1.0.0', '1.0.0:semver', true],
+  ['LESS_THAN_INCLUSIVE', '1.0.1', '1.0.0:semver', false]
+].freeze
+
+RSpec.describe Flagsmith::Engine::Segments::Condition do
+  TEST_CASES.each do |(operator, trait_value, condition_value, expected_result)|
+    it "#{operator}, #{condition_value} #match_trait_value #{trait_value}" do
+      segment_condition = Flagsmith::Engine::Segments::Condition.new(
+        operator: operator, property: 'foo', value: condition_value
+      )
+      expect(segment_condition.match_trait_value?(trait_value)).to eq(expected_result)
+    end
+  end
+end
+
+RSpec.describe Flagsmith::Engine::Segment do
+  it 'test_segment_schema_engine_model_object_to_dict(project)' do
+    # Given
+    segment = Flagsmith::Engine::Segment.new(
+      id: 1,
+      name: 'Segment',
+      rules: [
+        Flagsmith::Engine::Segments::Rule.new(
+          type: 'ALL_RULE',
+          conditions: [
+            Flagsmith::Engine::Segments::Condition.new(operator: 'EQUAL', property: 'foo', value: 'bar')
+          ]
+        )
+      ],
+      feature_states: [
+        Flagsmith::Engine::Features::State.new(
+          django_id: 1,
+          feature: Flagsmith::Engine::Feature.new(
+            id: 1,
+            name: 'my_feature',
+            type: 'STANDARD'
+          ),
+          enabled: true
+        )
+      ]
+    )
+
+    # When
+    data = segment
+
+    # Then
+    expect(data.feature_states.length).to eq(1)
+    expect(data.rules.length).to eq(1)
+  end
+
+  # it 'test_dict_to_segment_model' do
```

Comment:
How come this code is commented out?

Reply:
I planned to delete these tests, because I don’t see any use in them, this logic is covered by other tests